hail/python/test/hail/methods/test_impex.py

```diff
@@ -930,7 +930,7 @@ def test_export_gen_exprs(self):
         in1 = (hl.import_gen(out1 + '.gen', sample_file=out1 + '.sample', min_partitions=3)
                .add_col_index()
                .add_row_index())
-        self.assertTrue(in1.aggregate_entries(hl.agg.fraction(in1.GP == [0.0, 1.0, 0.0])) == 1.0)
+        self.assertTrue(in1.aggregate_entries(hl.agg.fraction((hl.is_missing(in1.GP) | (in1.GP == [0.0, 1.0, 0.0])) == 1.0)))
```

Comment:
This change switches a `select_entries` to a `annotate_entries / drop`, which removes the implicit entry unfiltering. As a consequence, the semantics of the following:

```
         hl.export_gen(gen, out1, id1=hl.str(gen.col_idx), id2=hl.str(gen.col_idx), missing=0.5,
                       varid=hl.str(gen.row_idx), rsid=hl.str(gen.row_idx), gp=[0.0, 1.0, 0.0])
```

has changed. In **current master**, this will export GP=[0,1,0] at **every entry, regardless of missingness**. In this branch, it will respect missingness.

I consider this a bugfix, but I want input from the group.

Reply:
> removes the implicit entry unfiltering

I strongly feel this was a bug and needs to be fixed.